src/main/java/com/gw2auth/oauth2/server/service/gw2account/apitoken/Gw2AccountApiTokenServiceImpl.java

```diff
@@ -146,20 +145,26 @@ public void updateApiTokensValid(Instant lastValidCheckTime, Collection<Gw2Accou
     @Scheduled(fixedRate = 10L, timeUnit = TimeUnit.MINUTES)
     public void checkTokenValidity() {
         final Instant now = this.clock.instant();
-        final List<Gw2AccountApiTokenEntity> tokensToCheck = this.gw2AccountApiTokenRepository.findAllByLastValidTimeGTEAndLastValidCheckTimeLTE(
+        final List<Gw2AccountApiTokenValidCheckEntity> tokensToCheck = this.gw2AccountApiTokenRepository.findAllByLastValidTimeGTEAndLastValidCheckTimeLTE(
                 now.minus(IGNORE_TOKENS_INVALID_FOR_LONGER_THAN),
                 now.minus(VALIDITY_CHECK_INTERVAL),
                 VALIDITY_CHECK_BATCH_SIZE
         );
-        final List<Gw2AccountApiTokenValidUpdateEntity> updateEntities = new ArrayList<>(tokensToCheck.size());
-        final int[] counts = new int[3];
+        final List<Gw2AccountApiTokenValidUpdateEntity> validUpdateEntities = new ArrayList<>(tokensToCheck.size());
+        final List<Gw2AccountNameUpdateEntity> nameUpdateEntities = new ArrayList<>(tokensToCheck.size());
+        final int[] counts = new int[4];
```

Comment:
Please replace this with actual variables (e.g. `int validCount = 0, invalidCount = 0, entityCount = 0, unknownCount = 0,`) or at least give the counts array a comment with that information. There is no reason for this to be an array. 😅

Reply:
This is still an artifact from when the counts were updated inside lambda expressions, thus requiring a final variable. Fixed that :)